File: apidemo/sensordemo/environmentalsensordemo/src/main/java/com/gdkdemo/sensor/environmental/cp/core/Log.java
package com.gdkdemo.sensor.environmental.cp.core;

import android.os.SystemClock;


final class Log 
{
    static final String LOGTAG = "GdkDemoSensorDemo.cp.core";
    private static final int LOGLEVEL = Config.LOGLEVEL;

    static final boolean V = LOGLEVEL <= android.util.Log.VERBOSE;
    static final boolean D = LOGLEVEL <= android.util.Log.DEBUG;
    static final boolean I = LOGLEVEL <= android.util.Log.INFO;
    static final boolean W = LOGLEVEL <= android.util.Log.WARN;
    static final boolean E = LOGLEVEL <= android.util.Log.ERROR;

    static void v(String logMe) 
    {
        android.util.Log.v(LOGTAG, SystemClock.uptimeMillis() + " " + logMe);
    }
    static void v(String logMe, Throwable ex)
    {
        android.util.Log.v(LOGTAG, SystemClock.uptimeMillis() + " " + logMe, ex);
    }

    static void d(String logMe) 
    {
        android.util.Log.d(LOGTAG, SystemClock.uptimeMillis() + " " + logMe);
    }
    static void d(String logMe, Throwable ex)
    {
        android.util.Log.d(LOGTAG, SystemClock.uptimeMillis() + " " + logMe, ex);
    }

    static void i(String logMe)
    {
        android.util.Log.i(LOGTAG, logMe);
    }
    static void i(String logMe, Throwable ex)
    {
        android.util.Log.i(LOGTAG, logMe, ex);
    }

    static void w(String logMe)
    {
        android.util.Log.w(LOGTAG, logMe);
    }
    static void w(String logMe, Throwable ex)
    {
        android.util.Log.w(LOGTAG, logMe, ex);
    }

    static void e(String logMe)
    {
        android.util.Log.e(LOGTAG, logMe);
    }
    static void e(String logMe, Exception ex)
    {
        android.util.Log.e(LOGTAG, logMe, ex);
    }
}

File: xamarindemo/VoiceDemo/Log.cs
using System;
using SystemClock = Android.OS.SystemClock;
using Java.Lang;
using VoiceDemo.Common;

namespace VoiceDemo
{
	internal static class Log 
	{
		private const string TAG = "VoiceDemo";

		internal static readonly bool V = Config.LogLevel <= Android.Util.Log.Verbose;
		internal static readonly bool D = Config.LogLevel <= Android.Util.Log.Debug;
		internal static readonly bool I = Config.LogLevel <= Android.Util.Log.Info;
		internal static readonly bool W = Config.LogLevel <= Android.Util.Log.Warn;
		internal static readonly bool E = Config.LogLevel <= Android.Util.Log.Error;

		static void v(string logMe)
		{
			Android.Util.Log.v(TAG, SystemClock.UptimeMillis() + " " + logMe);
		}
		static void v(string logMe, Throwable ex)
		{
			Android.Util.Log.v(TAG, SystemClock.UptimeMillis() + " " + logMe, ex);
		}

		static void d(string logMe) 
		{
			Android.Util.Log.d(TAG, SystemClock.UptimeMillis() + " " + logMe);
		}
		static void d(string logMe, Throwable ex)
		{
			Android.Util.Log.d(TAG, SystemClock.UptimeMillis() + " " + logMe, ex);
		}

		static void i(string logMe)
		{
			Android.Util.Log.i(TAG, logMe);
		}
		static void i(string logMe, Throwable ex)
		{
			Android.Util.Log.i(TAG, logMe, ex);
		}

		static void w(string logMe)
		{
			Android.Util.Log.w(TAG, logMe);
		}
		static void w(string logMe, Throwable ex)
		{
			Android.Util.Log.w(TAG, logMe, ex);
		}

		static void e(string logMe)
		{
			Android.Util.Log.e(TAG, logMe);
		}
		static void e(string logMe, System.Exception ex)
		{
			Android.Util.Log.e(TAG, logMe, ex);
		}
	}
}
